source/fundamentals/crud/write-operations/pkFactory.txt

```diff
@@ -0,0 +1,99 @@
+.. _node-pkfactory:
+
+===================
+Primary Key Factory
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about the primary key factory features
+and limitations in the MongoDB Node.js driver. 
+
+Behavior
+--------
+
+A primary key factory is a class that defines a system for creating
+unique identifiers for documents and implements  the ``pkFactory``
+interface. 
```

Comment:
issue: This is great content but I think it's a bit out of order. If the page is about how to generate custom _id values (primary key), it might be better to go that route initially.

> In this guide, you can learn how to generate your own custom ``_id`` values using the driver's **pkFactory** option. <now define pkFactor>.